ui/components/ui/custom-spending-cap/custom-spending-cap.js

```diff
@@ -0,0 +1,160 @@
+import FormField from '../form-field';
+import Typography from '../typography';
+import {
+  ALIGN_ITEMS,
+  COLORS,
+  DISPLAY,
+  FLEX_DIRECTION,
+  FONT_WEIGHT,
+  TYPOGRAPHY,
+  JUSTIFY_CONTENT,
+  SIZES,
+} from '../../../helpers/constants/design-system';
+
+export default function CustomSpendingCap({
+  tokenName,
+  currentTokenBalance,
+  dappProposedValue,
+}) {
+  const t = useContext(I18nContext);
+  const [value, setValue] = useState('');
+  const [customSpendingCapText, setCustomSpendingCapText] = useState('');
+  const [error, setError] = useState('');
+  const inputLogicEmptyStateText = t('inputLogicEmptyState');
+
+  const getInputTextLogic = (inputNumber) => {
+    if (inputNumber <= currentTokenBalance) {
+      return {
+        className: 'custom-spending-cap__lowerValue',
+        description: t('inputLogicEqualOrSmallerNumber', [
+          <Typography
+            key="custom-spending-cap"
+            variant={TYPOGRAPHY.H6}
+            fontWeight={FONT_WEIGHT.BOLD}
+            className="custom-spending-cap__input-value-and-token-name"
+          >
+            {inputNumber} {tokenName}
+          </Typography>,
+        ]),
+      };
+    } else if (inputNumber > currentTokenBalance) {
+      return {
+        className: 'custom-spending-cap__higherValue',
+        description: t('inputLogicHigherNumber'),
+      };
+    }
+    return {
+      className: 'custom-spending-cap__emptyState',
+      description: t('inputLogicEmptyState'),
+    };
+  };
+
+  const handleChange = (valueInput) => {
+    let spendingCapInvalid = '';
+    const inputTextLogic = getInputTextLogic(valueInput);
+    const inputTextLogicDescription = inputTextLogic.description;
+
+    if (valueInput < 0 || isNaN(valueInput)) {
+      spendingCapInvalid = t('spendingCapInvalid');
+      setCustomSpendingCapText('');
+      setError(spendingCapInvalid);
+    } else {
+      setCustomSpendingCapText(inputTextLogicDescription);
+      setError('');
+    }
+
+    setValue(valueInput);
+  };
+
+  const chooseTooltipContentText =
+    value > currentTokenBalance
+      ? t('warningTooltipText', [
+          <Typography
+            key="tooltip-text"
+            variant={TYPOGRAPHY.H7}
+            fontWeight={FONT_WEIGHT.BOLD}
+            color={COLORS.WARNING_DEFAULT}
+          >
+            <i className="fa fa-exclamation-circle" /> {t('beCareful')}
+          </Typography>,
+        ])
+      : t('inputLogicEmptyState');
+
+  return (
+    <Box
+      className="custom-spending-cap"
+      borderRadius={SIZES.SM}
+      paddingTop={4}
+      paddingRight={2}
+      paddingLeft={2}
+      display={DISPLAY.FLEX}
+      alignItems={ALIGN_ITEMS.FLEX_START}
+      flexDirection={FLEX_DIRECTION.COLUMN}
+      backgroundColor={COLORS.BACKGROUND_ALTERNATIVE}
+      gap={2}
+    >
+      <Box
+        justifyContent={JUSTIFY_CONTENT.CENTER}
+        marginTop={4}
+        marginRight={6}
+        marginLeft={6}
+        marginBottom={4}
```

Comment:
Can this not be added to the padding props in the parent `Box` component above?

Reply:
@georgewrmarshall Done [here](https://github.com/MetaMask/metamask-extension/pull/15522/commits/190f93d7a3b816927c6d40f7039080fdc87be55b).